src/pages/videos/VideoSearch.tsx

```diff
@@ -92,6 +100,13 @@ const VideoSearch = () => {
         searchResultActions.refetch();
     };
 
+    const videoSelection = useResultListSelect();
+    createEffect(() => {
+        if (searchResults.loading) {
+            videoSelection.clear();
+        }
+    });
```

Comment:
TODO: better approach for this.

First of all, this clears the selection on pagination change, which is just wrong.  
But it's kind of fragile in general.